src/freenet/node/diagnostics/DefaultNodeDiagnostics.java

```diff
@@ -0,0 +1,46 @@
+/* This code is part of Freenet. It is distributed under the GNU General
+ * Public License, version 2 (or at your option any later version). See
+ * http://www.gnu.org/ for further details of the GPL. */
+package freenet.node.diagnostics;
+
+import freenet.node.diagnostics.threads.*;
+import freenet.support.Ticker;
+import freenet.node.NodeStats;
+import freenet.node.NodeInitException;
+
+import java.util.HashMap;
+
+/**
+ *  @author desyncr
+ *
+ *  A class to retrieve data to build diagnostic dumps to help in determining
+ *  node bottlenecks or misconfiguration.
+ *
+ *  This class launches various threads at intervals to retrieve information. This information
+ *  is available through the public methods.
+ *  Some data pointers are obtained from NodeStats object.
+ */
+public class DefaultNodeDiagnostics implements NodeDiagnostics {
+    private final DefaultThreadDiagnostics defaultThreadDiagnostics;
+
+   /**
+     * @param nodeStats Used to retrieve data points.
+     * @param ticker Used to queue timed jobs.
+     */
+    public DefaultNodeDiagnostics(final NodeStats nodeStats, final Ticker ticker) {
```

Comment:
These `final`s here are only cluttering up the view and don’t do anything, please remove.